src/test/java/com/hubspot/jinjava/lib/filter/IntersectFilterTest.java

```diff
@@ -31,4 +34,35 @@ public void itReturnsEmptyOnNullParameters() {
     assertThat(jinjava.render("{{ [1, 2, 3]|intersect(null) }}", new HashMap<>()))
       .isEqualTo("[]");
   }
+
+  @Test
+  public void itDoesNotThrowWarningOnMatchedTypes() {
+    JinjavaInterpreter interpreter = jinjava.newInterpreter();
+
+    String renderedOutput = interpreter.render("{{ [1, 2, 3]|intersect([1, 2, 3]) }}");
+    assertThat(renderedOutput).isEqualTo("[1, 2, 3]");
+
+    List<TemplateError> errors = interpreter.getErrors();
+    assertThat(errors).isEmpty();
+  }
+
+  @Test
+  public void itThrowsWarningOnMismatchTypes() {
```

Comment:
Can we add a unit test for the empty set case to make sure it doesn't add a warning.